### PR TITLE
feat: sql runner monaco editor

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: [#10615](https://github.com/lightdash/lightdash/issues/10615)

### Description:

Use monaco editor and support all the warehouse syntax.
Screenshot 2024-07-09 at 14 58 53

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by joaoviana** [comment]
Just a note when we load the monaco editor. I get this on load (I believe the same happens on `custom viz`)
image

```
undefined:1 Uncaught SyntaxError: Unexpected token '<' (at undefined:1:1)
```

**Comment by ZeRego** [comment]
> Just a note when we load the monaco editor. I get this on load (I believe the same happens on `custom viz`) image
> 
> ```
> undefined:1 Uncaught SyntaxError: Unexpected token '<' (at undefined:1:1)
> ```

Still haven't figured out where/why this happens.

File: packages/frontend/src/features/sqlRunner/components/SqlEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    onSqlChange: (value: string) => void;
+}> = ({ sql, onSqlChange }) => {
+    const projectUuid = useAppSelector((state) => state.sqlRunner.projectUuid);
+    const { data, isLoading } = useProject(projectUuid);
+
+    const language = useMemo(
+        () => getLanguage(data?.warehouseConnection?.type),
+        [data],
+    );
+    const onMount = useCallback(
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/components/SqlEditor.tsx:89]: could you rename to `beforeMount`?

**Review by joaoviana — APPROVED** [review]
Works great!

**Pull request merged**